src/render_cmd.rs

```diff
@@ -287,6 +312,30 @@ impl RenderCmd {
         }
         Ok(buffer)
     }
+
+    ///  Filter out the deployment values that match the regex
+    fn filter_deployment(&self, regex: String, name: &String, values: Option<Vec<PathBuf>>) -> Option<Vec<PathBuf>> {
+        let is_name_filtered = self.is_name_filtered(regex, &name);
+
+        if is_name_filtered {
+            return values;
+        }
+        return Some(vec![]);
+    }
+
+    
+
+    /// Check if deployment name is filtered
+    fn is_name_filtered(&self, regex: String, name: &String) -> bool {
+        let regex_string = format!("[{}]{}", name, "{4,}");
```

Comment:
Please allow the user to define the regular expression using `--filter`.

Reply:
It is a good opportunity to add unit tests. For example given a list of deployments:

```yaml
- name: 'edge-eu-w4'
- name: 'stage-eu-w4'
- name: 'prod-as-e1'
- name: 'prod-eu-w4'
- name: 'prod-us-c1'
```

Given `--filter edge` or, move specific, `--filter ^edge` should both result in only the edge deployment rendered.
Given `--filter prod`, should only have the three prod deployment rendered

You could also test for a regex that matches on everything in `eu-w4`, for example.